app/Integration/Article/Repository/ArticleQueryBuilderRepository.php

```diff
@@ -37,7 +36,7 @@
      */
     private $logger;
 
-    public function __construct(DB $database, ArticleMapper $articleMapper, LoggerInterface $logger)
+    public function __construct(DatabaseManager $database, ArticleMapper $articleMapper, LoggerInterface $logger)
```

Comment:
I think you are confusing SRP with wth else.

It the repository was also creating methods to logs it was breaking the SRP, otherwise, in your point of you all the objects that are using more than one dependency are violating the SRP 😅